src/shared/tpm2-util.h

```diff
@@ -73,15 +76,25 @@ int tpm2_get_good_pcr_banks_strv(ESYS_CONTEXT *c, uint32_t pcr_mask, char ***ret
 
 int tpm2_extend_bytes(ESYS_CONTEXT *c, char **banks, unsigned pcr_index, const void *data, size_t data_size, const void *secret, size_t secret_size);
 
-#else
+int tpm2_list_devices(void);
+int tpm2_find_device_auto(int log_level, char **ret);
+
+#else /* HAVE_TPM2 */
+
 struct tpm2_context;
-#endif
 
-int tpm2_context_init(const char *device, struct tpm2_context *ret);
-void tpm2_context_destroy(struct tpm2_context *c);
+static inline int tpm2_not_supported(void) {
+        return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "TPM2 not supported on this build.");
+}
+#define tpm2_context_init(...) tpm2_not_supported()
+#define tpm2_context_destroy(...) tpm2_not_supported()
```

Comment:
hmm, we usually repeat the prototypes as "static inline" stubs, to ensure types at least are checked. This actually matters since a good chunk of the CI disables various deps when analyzing, hence any type info we'll get even with deps off is *good*

Reply:
(also, if people call these functions with parameters with side-effects these definitions will remove the side-effects. i.e. if i call `tpm2_context_destroy(TAKE_PTR(c))` then the magic effect of TAKE_PTR() will be voided...

Reply:
the duplication really bugs me, but it's not important for the overall tpm2 work; maybe i'll try to think of something as a separate PR. Will drop it in next push.